src/main/java/io/appform/dropwizard/sharding/dao/ILookupDao.java

```diff
@@ -0,0 +1,52 @@
+package io.appform.dropwizard.sharding.dao;
+
+import io.appform.dropwizard.sharding.config.ShardingBundleOptions;
+import io.appform.dropwizard.sharding.dao.LookupDao.ReadOnlyContext;
+import io.appform.dropwizard.sharding.utils.ShardCalculator;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import org.hibernate.Session;
+import org.hibernate.criterion.DetachedCriteria;
+
+public interface ILookupDao<T> {
```

Comment:
We can do without `I` prefix. Interfaces should be indistinguishable from their implementation and any user of the interface should not be concerned that they are working with an interface. 

For implementation, use `LookupDaoImpl` or `LookupDaoDefault` or some name which aligns more closely with the existing implementation